fs/glob.ts

```diff
@@ -97,47 +159,187 @@ export interface ExpandGlobOptions extends GlobOptions {
 // This is a very incomplete solution. The whole directory tree from `root` is
 // walked and parent paths are not supported.
 export async function* expandGlob(
-  globString: string,
+  glob: string,
   {
     root = cwd(),
+    exclude = [],
     includeDirs = true,
     extended = false,
-    globstar = false,
-    strict = false,
-    filepath = true,
-    flags = ""
+    globstar = false
   }: ExpandGlobOptions = {}
 ): AsyncIterableIterator<WalkInfo> {
-  const absoluteGlob = isAbsolute(globString)
-    ? globString
-    : join(root, globString);
-  const globOptions = { extended, globstar, strict, filepath, flags };
-  yield* walk(root, {
-    match: [glob(absoluteGlob, globOptions)],
-    includeDirs
-  });
+  const resolveFromRoot = (path: string): string =>
+    isAbsolute(path) ? normalize(path) : join(root, path);
+  const globOptions: GlobOptions = { extended, globstar };
+  const excludePatterns = exclude
+    .map(resolveFromRoot)
+    .map((s: string): RegExp => globToRegExp(s, globOptions));
+  const shouldInclude = ({ filename }: WalkInfo): boolean =>
+    !excludePatterns.some((p: RegExp): boolean => !!filename.match(p));
+  const { segments, hasTrailingSep, winRoot } = split(resolveFromRoot(glob));
+
+  let fixedRoot = winRoot != undefined ? winRoot : "/";
+  while (segments.length > 0 && !isGlob(segments[0])) {
+    fixedRoot = join(fixedRoot, segments.shift()!);
+  }
+
+  let fixedRootInfo: WalkInfo;
+  try {
+    fixedRootInfo = { filename: fixedRoot, info: await stat(fixedRoot) };
+  } catch {
+    return;
```

Comment:
Is this swallowing of the error intentional? I think it's better to throw here.

Reply:
It's intentional. If the fixed root doesn't exist we simply know in advance that there are no matches. I think that's the expected behaviour of glob expansion.